### PR TITLE
feat: Backport RPM packaging to 1.x branch

### LINKED ISSUE #4402: [feat] Bundle Tauri apps as RPM
### Describe the problem

It would be nice to have rpm packages for Tauri apps as there is no Flatpak support yet and alien does not manage to convert the deb packages (at least in my personal attempts).

### Describe the solution you'd like

Bundling the application to an rpm package just like the deb packaging is done right now.

### Alternatives considered

Flatpaks are somewhat of an alternative, but I believe actual rpm packages would be great, especially because we get the advantage of the really small bundle size of Tauri apps, unlike AppImages and Flatpak alike.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hjmallon**
feat(bundler): Add RPM packaging, closes #4402 (#5202)

* feat(bundler): Add RPM packaging

* feat(bundler): Update 'rpm' to 0.13.1

* Fix fmt

------------

This is a backport of https://github.com/tauri-apps/tauri/pull/5202 to the v1 branch

### CODE REVIEW

File: tooling/bundler/src/bundle/linux/rpm.rs [hunk 1]
```diff
@@ -5,10 +5,147 @@
...
+      .with_context(|| format!("Failed to remove old {package_base_name}"))?;
+  }
+  fs::create_dir_all(&package_dir)?;
+  let package_path = base_dir.join(&package_name);
+
+  info!(action = "Bundling"; "{} ({})", package_name, package_path.display());
+
+  let mut builder = rpm::PackageBuilder::new(name, version, license, arch, summary)
+    .epoch(epoch)
+    .release(release);
```
Comment (FabianLars) [inline tooling/bundler/src/bundle/linux/rpm.rs]: Can you add this in this PR too? https://github.com/tauri-apps/tauri/pull/9840/files

File: tooling/bundler/Cargo.toml [hunk 2]
```diff
@@ -65,6 +65,7 @@ regex = "1"
 heck = "0.5"
 ar = "0.9.0"
 md5 = "0.7.0"
+rpm = "0.13.1"
```
Comment (FabianLars) [inline tooling/bundler/Cargo.toml]: 0.14 was released some time after the v2 pr, we might as well upgrade it here already.

File: .changes/bundler-rpm.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"tauri-bundler": 'patch:enhance'
+"tauri-cli": 'patch:enhance'
+"@tauri-apps/cli": 'patch:enhance'
```
Comment (FabianLars) [inline .changes/bundler-rpm.md]: i _think_ we need `tauri-utils` here too.

Comment (FabianLars) [inline .changes/bundler-rpm.md]: and they also have to be minor not patch. They were only patch in v2 because v2 is still in prerelease.

**Review by FabianLars — CHANGES_REQUESTED** [review]
Thank you! This time i'm quicker to review your PR 😅 

Just a few minor things i could have added myself but since i'm waiting on https://github.com/tauri-apps/tauri/pull/9796#discussion_r1608172756 we still have some time

File: .changes/bundler-rpm.md (hunk 3 above)
Comment (hjmallon) [reply .changes/bundler-rpm.md]: Sorry I'm not totally clear what to put here, the CI says I did it wrong. Should it be 'minor:feat'?

Comment (FabianLars) [reply .changes/bundler-rpm.md]: yep, minor:feat should work

**Comment by hjmallon** [comment]
@FabianLars : One of the CI tests is failing for me, but I think it is because I get the following diff on this branch atm after building, it seems unrelated to my change so I dont think I should integrate it?

```
diff --git a/Cargo.lock b/Cargo.lock
index bab6a8118..3e5aed09a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -4021,7 +4021,7 @@ checksum = "e1fc403891a21bcfb7c37834ba66a547a8f402146eba7265b5a6d88059c9ff2f"
 
 [[package]]
 name = "tauri"
-version = "1.6.5"
+version = "1.6.6"
 dependencies = [
  "anyhow",
  "base64 0.21.7",
```

**Comment by FabianLars** [comment]
We'll probably merge another pr before yours which will include this change. If you want to have green ci already you can just include this change. It's just some annoyance of our publishing pipeline :/

**Review by FabianLars — APPROVED** [review]
I didn't test it yet because I'm still not home (send help pls) but considering it's a backport, seeing that CI is happy and with my comments being resolved I think this is good to go.

**Comment by FabianLars** [comment]
P.S. Big thanks for the contribution :)

**Comment by FabianLars** [comment]
P.P.S. I swear I'll check your dmg backport when I'm back x)

**Comment by hjmallon** [comment]
> P.P.S. I swear I'll check your dmg backport when I'm back x)

Thanks, I will rebase it tomorrow ready to go

**Pull request merged**